src/MonitorProfiler/MainProfiler/MainProfiler.cpp

```diff
@@ -22,21 +24,37 @@ GUID MainProfiler::GetClsid()
 
 STDMETHODIMP MainProfiler::Initialize(IUnknown *pICorProfilerInfoUnk)
 {
+    ExpectedPtr(pICorProfilerInfoUnk);
+
     HRESULT hr = S_OK;
 
-    if (FAILED(hr = ProfilerBase::Initialize(pICorProfilerInfoUnk)))
-    {
-        std::cerr << "Failed to initialize profiler." << std::endl;
-    }
+    IfFailRet(ProfilerBase::Initialize(pICorProfilerInfoUnk));
+    IfFailRet(InitializeEnvironment());
+    IfFailRet(InitializeLogging());
+
+    // Logging is initialized and can now be used
 
-    IfFailRet(m_pCorProfilerInfo->SetEnvironmentVariable(
-        _T("DOTNETMONITOR_ProductVersion"),
-        MonitorProductVersion_TSTR
-        ));
+    // Set product version environment variable to allow discovery of if the profiler
+    // as been applied to a target process. Diagnostic tools must use the diagnostic
+    // communication channel's GetProcessEnvironment command to get this value.
+    IfFailLogRet(EnvironmentHelper::SetProductVersion(m_pEnvironment, m_pLogger));
+
+#ifdef WIN32
```

Comment:
I've been using TARGET_WINDOWS. Is there a preferred set of defines here?

Reply:
`TARGET_*` sounds good to me